core/src/main/java/com/netflix/iceberg/RemoveSnapshots.java

```diff
@@ -115,7 +115,10 @@ public void commit() {
         .onlyRetryOn(CommitFailedException.class)
         .run(item -> {
           TableMetadata updated = internalApply();
-          ops.commit(base, updated);
+          // only commit the updated metadata if at least one snapshot was removed
+          if (updated.snapshots().size() != base.snapshots().size()) {
```

Comment:
Would it be more precise to check for `if (updated.snapshots().size() < base.snapshots().size())`?

Reply:
I think those two checks are functionally equivalent for this check.  This is to effectively check that the number of snapshots has not changed.  I would err on the side of equality as the proposed change would conceptually mean that you could increase the number of snapshots, which would not be expected in this case.